exonum/tests/collections/proof_map_index.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// cspell:ignore oneof
+
+//! Property testing for proof map index as a rust collection.
+
+use exonum::storage::{Database, Fork, MemoryDB, ProofMapIndex, StorageValue};
+use modifier::Modifier;
+use proptest::{collection::vec, num, prelude::*, strategy, test_runner::TestCaseResult};
+
+use std::collections::HashMap;
+
+use super::{MapAction, ACTIONS_MAX_LEN};
+
+impl<'a, V> Modifier<ProofMapIndex<&'a mut Fork, [u8; 32], V>> for MapAction<[u8; 32], V>
+where
+    V: StorageValue,
+{
+    fn modify(self, map: &mut ProofMapIndex<&mut Fork, [u8; 32], V>) {
+        match self {
+            MapAction::Put(mut k, v) => {
+                k[0] = k[0] % 8;
+                map.put(&k, v);
+            }
+            MapAction::Remove(mut k) => {
+                k[0] = k[0] % 8;
+                map.remove(&k);
+            }
+            MapAction::Clear => {
+                map.clear();
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+fn compare_collections(
```

Comment:
Perhaps the compare should take into account the sorting?

Can we replace `HashMap` by `BTreeMap` and checks that iterators compatible? (probably we could reuse `compare_collections` method as well)

Reply:
Is it important to have sorting for proof map or it is just implementation detail?